scripts/json2csv.mjs

```diff
@@ -0,0 +1,172 @@
+    input.metrics.megabytes_per_second_boost.values.med, // MB/s Med
+    input.metrics.megabytes_per_second_boost.values.max, // MB/s Max
+    input.metrics.megabytes_per_second_boost.values['p(90)'], // MB/s P(90)
+    input.metrics.megabytes_per_second_boost.values['p(95)'], // MB/s P(95)
+    input.metrics.success_boost.values.rate, // Success Rate
+    '', // Missed WindowPost
+    '', // CPU
+    '' // Memory
+  ]
+
+  return [https, boost]
+}
+
+/**
+ * Turn an array into a line of CSV data. Currently this doesn't need to do
+ * anything fancy but quoting or escaping may be required as the data evolves.
+ * e.g. scenarios get quotes or commas in their names or numbers start to print
+ * with scientific notation or some form that isn't easily imported as a
+ * spreadsheet format.
+ *
+ * @param {any[]} data
+ * @returns {string}
+ */
+function toCSV (data) {
+  return data.join(',')
+}
+
+/**
+ * filenameSort is a custom sort function for the test data files. It sorts the
+ * files by the number at the start of the filename, then optionally by the byte
+ * size it's also in the name (as it is for range-requests).
+ * We sort files to ensure that the rows in the CSV are in a reasonable order.
+ *
+ * @param {{name: string}} a
+ * @param {{name: string}} b
+ * @returns {number}
+ */
+function filenameSort (a, b) {
+  // extract the VU from the start of the filename, then optionally the
+  // byte size as the second element
+  const valuesFromname = (name) => {
+    const m = name.match(/^(\d+)vu_(?:(\d+)B)?/)
+    return [parseInt(m[1], 10), parseInt(m[2], 10)]
+  }
+  const ai = valuesFromname(a.name)
+  const bi = valuesFromname(b.name)
+  // sort by VU first, then byte size if VU is the same
+  if (ai[0] === bi[0]) {
+    return ai[1] === bi[1] ? 0 : ai[1] < bi[1] ? -1 : 1
+  }
+  return ai[0] < bi[0] ? -1 : 1
+}
+
+/**
+ * The main entry point for the script. This reads all the JSON files from the
+ * `out` directory and transforms them into a single CSV file.
+ */
+async function run () {
+  // ingest all the JSON data into a single array
+  let data = [header]
+  const outDir = new URL('../out/', import.meta.url)
+  for (const dir of await fs.readdir(outDir, { withFileTypes: true })) {
+    const test = dir.name
+    if (!dir.isDirectory()) {
+      continue
+    }
+    const root = new URL(`./${test}/`, outDir)
+    let files = await fs.readdir(root, { withFileTypes: true })
+    // sort the files so they end up in our output in a nice order
+    files = files.filter((f) => f.isFile()).sort(filenameSort)
+    for (const file of files) {
+      const name = file.name.replace(/\.json$/)
+      const input = JSON.parse(await fs.readFile(new URL(file.name, root)))
+      data = data.concat(processInput(test, name, input))
+    }
+  }
+
+  // transform the data into CSV
+  const contents = data.map(toCSV).join('\n') + '\n'
+  const out = process.argv[2] || new URL('../out/results.csv', import.meta.url).pathname
```

Comment:
as long as we're doing that, I wouldn't mind providing the same name from loadtest.sh into the script so they're all the same

Reply:
Have a look at the latest commit, I set the date once in loadtest.sh and provide it through the runs; I truncated the seconds off it because that seems redundant once we're doing it across runs. Also doing it in UTC.

The downside of this is that you don't capture the true time of the test if they are taking a long time. Maybe there's a reason we'd find that relevant?

but then we come back to the original problem, what is the relevant part for the results CSV name?

Reply:
The only thing I can think of is debugging. If we're trying to correlate individual test runtimes with say, a trace in boost, it may make it a bit more difficult. I just noticed the times are not in the CSV output. I find value in the individual test times, but I feel maybe test start time is relevant for the CSV only for the purpose of uniquely identifying test results.